objects/oplayercontroller/Step_0.gml

```diff
@@ -558,6 +618,76 @@ if (!isGrabbed)
 // Handle Enviornmental Displacement
 environmentDisplacement = 0;
 
+// Handle freezing screen
+if (state == eState.SCREEN_FREEZE)
+{
+	image_speed = 0;
+	hsp = 0;
+	environmentDisplacement = 0;
+	vsp = 0;
+	
+	// If player is performing Rush Cancel
+	if (rcActivated)
+	{
+		// Screen freeze for Rush Cancel lasts for 30 frames
+		if (rcFreezeTimer >= 30)
+		{
+			rcActivated = false;
+			rcFreezeTimer = 0;
+			activateFreeze = false;
+			global.freezeTimer = false;
+			animTimer = 0; // Reset the animation timer when entering Rush Cancel state
+			speedTrailTimer = 0;
+			comboScaling += 3.0;
+			if (!grounded)
+			{
+				state = eState.RUSH_CANCEL_AIR;
+			}
+			else if (verticalMoveDir == 1)
+			{
+				vsp = -global.rcUpSpeed;
+				jumpHsp = walkSpeed * 1.5 * image_xscale;
+				state = eState.RUSH_CANCEL_UP;
+				grounded = false;
+			}
+			else
+			{
+				rcForwardTimer = 0;
+				state = eState.RUSH_CANCEL_FORWARD;
+			}
+		}
+		else
+		{
+			rcFreezeTimer++;

```

Comment:
If Colby is going to work on the Pause Menu, then it would make sense for that menu to also use the screen freeze to make sure everything stops. Because the player could keep the game paused for as long as they want, the system needs to have some way of deactivating screen freeze on command. It's maybe not something you need to do right now, but it might be something that Colby brings up.